pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -299,13 +298,28 @@ func (lbaas *LbaasV2) GetLoadBalancer(clusterName string, service *api.Service)
 	return status, true, err
 }
 
+func nodeAddressForLB(node *api.Node) (string, error) {
+	addrs := node.Status.Addresses
+	if len(addrs) == 0 {
+		return "", ErrNoAddressFound
+	}
+
+	for _, addr := range addrs {
+		if addr.Type == api.NodeInternalIP {
```

Comment:
Hmm, seems somewhat circuituous to consult the node object for an ip we (cloudprovider) are responsible for populating in the first place, but I can live with it.